packages/api/src/graphql/article.ts

```diff
@@ -128,6 +128,20 @@ export const GraphQLArticleRevision = new GraphQLObjectType<ArticleRevision, Con
 
     canonicalUrl: {type: GraphQLString},
 
+    url: {
+      type: GraphQLNonNull(GraphQLString),
+      resolve: createProxyingResolver(({slug}, args, {urlAdapter}, info) => {
+        const id = info.variableValues.id
+
+        if (!slug) return ''
+
+        return urlAdapter.getPublicArticleURL({
```

Comment:
you need to pass the whole article revision. It's up to the we.publish operators how they build the URL.

Reply:
but `getPublicArticleURL` expecting `PublicArticle` fields. Should passing the revision is correct?

Reply:
i think you have all the necessary fields to create the public article. Only the publishedAt, publishAt and updatedAt fields might be missing. But those you can just set to now. because nobody should make an URL based on those dates.